src/index.js

```diff
@@ -33,8 +23,60 @@ items.forEach((item) => {
 
   input.addEventListener('change', () => {
     updateCompleted(item, input);
-    saveLocalStorage(items);
+    saveLocalStorage(itemsArray);
   });
 
   todos.appendChild(div);
+  const vertical = div.querySelector('.vertical');
+  const trash = div.querySelector('.trash');
+  const edit = div.querySelector('.edit');
+  edit.addEventListener('change', () => {
+    itemsArray.forEach((item) => {
+      if (item.index.toString() === div.id) {
+        item.description = edit.value;
+      }
+    });
+    saveLocalStorage(itemsArray);
+  });
+  vertical.addEventListener('click', () => {
+    vertical.style.display = 'none';
+    trash.style.display = 'block';
+  });
+  trash.addEventListener('click', () => {
+    div.remove();
+    itemsArray = itemsArray.filter((something) => {
+      if (something.index.toString() === div.id) {
+        return false;
+      }
+      return true;
+    });
+    saveLocalStorage(itemsArray);
+  });
+});
+
+const form = document.getElementById('form-id');
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+  const result = form.querySelector('input').value;
+  itemsArray.push({ completed: false, description: result, index: itemsArray.length });
+  saveLocalStorage(itemsArray);
+  window.location.reload();
```

Comment:
- [ ] It would be so much better if the `index` is set to the value of the new array length. Currently, the index starts from `0` and it must start from `1`. Please fix this minor issue as is one of the requirements. 
😄 💯